servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/apple/traffic/resilience/http/TrafficResilienceHttpClientFilter.java

```diff
@@ -0,0 +1,555 @@
+         * If {@code true} then the request will be {@link RequestRejectedException rejected}.
+         * <p>
+         * <b>It's important that instances returned from this {@link Function mapper} are singletons and shared
+         * across the same matched partitions. Otherwise, capacity will not be controlled as expected, and there
+         * is also the risk for {@link OutOfMemoryError}.</b>
+         *
+         * @param capacityPartitionsSupplier A {@link Supplier} to create a new {@link Function} for each new filter
+         * created by this {@link StreamingHttpClientFilterFactory factory}.
+         * Function provides a {@link CapacityLimiter} instance for the given {@link HttpRequestMetaData}.
+         * @param rejectNotMatched Flag that decides what the filter should do when {@code partitions} doesn't return
+         * a {@link CapacityLimiter}.
+         * @return {@code this}
+         */
+        public Builder capacityPartitions(
+                final Supplier<Function<HttpRequestMetaData, CapacityLimiter>> capacityPartitionsSupplier,
+                final boolean rejectNotMatched) {
+            this.capacityPartitionsSupplier = requireNonNull(capacityPartitionsSupplier);
+            this.rejectWhenNotMatchedCapacityPartition = rejectNotMatched;
+            return this;
+        }
+
+        /**
+         * Classification in the context of capacity management allows for hints to the relevant
+         * {@link CapacityLimiter} to be influenced on the decision-making process by the class of the
+         * {@link HttpRequestMetaData request}.
+         * <p>
+         * An example of classification, could be health checks that need to be given preference and still allowed
+         * a permit even under stress conditions. Another case, could be a separation of reads and writes, giving
+         * preference to the reads will result in a more available system under stress, by rejecting earlier writes.
+         * <p>
+         * The behavior of the classification and their thresholds could be different among different
+         * {@link CapacityLimiter} implementations, therefore the use of this API requires good understanding of how
+         * the algorithm in use will react for the different classifications.
+         * <p>
+         * Classification works within the context of a single
+         * {@link #Builder(Supplier, boolean)} partition} and not universally in the filter.
+         * <p>
+         * It's worth noting that classification is strictly a hint and could be ignored by the
+         * {@link CapacityLimiter}.
+         * @param classifier A {@link Function} that maps an incoming {@link HttpRequestMetaData} to a
+         * {@link Classification}.
+         * @return {@code this}.
+         */
+        public Builder classifier(final Function<HttpRequestMetaData, Classification> classifier) {
```

Comment:
Is it expected that `classifier` may have an internal state? If yes, we should use `Supplier` as well. Otherwise, we risk sharing a state between multiple clients instances.

Reply:
Addressed in #2921